tools/src/main/java/com/example/core/broker/testkit/SubscriberStub.java

```diff
@@ -0,0 +1,58 @@
+package com.example.core.broker.testkit;
+
+import akka.Done;
+import akka.actor.ActorRef;
+import akka.stream.Materializer;
+import akka.stream.OverflowStrategy;
+import akka.stream.javadsl.Flow;
+import akka.stream.javadsl.Keep;
+import akka.stream.javadsl.Sink;
+import akka.stream.javadsl.Source;
+import com.lightbend.lagom.javadsl.api.broker.Subscriber;
+
+import java.util.concurrent.CompletionStage;
+
+public class SubscriberStub<T> implements Subscriber<T> {
+
+    private String topicId;
+    private String groupId;
+    private ActorRef topicBuffer;
+    private final Materializer materializer;
+
+    public SubscriberStub(String topicId, ActorRef topicBuffer, Materializer materializer) {
+        this(topicId, "default", topicBuffer, materializer);
+    }
+
+    public SubscriberStub(String topicId, String groupId, ActorRef topicBuffer, Materializer materializer) {
+        this.topicId = topicId;
+        this.groupId = groupId;
+        this.topicBuffer = topicBuffer;
+        this.materializer = materializer;
+    }
+
+    @Override
+    public Subscriber<T> withGroupId(String groupId) throws IllegalArgumentException {
+        return new SubscriberStub<>(topicId, groupId, topicBuffer, materializer);
+    }
```

Comment:
Implements support for consumer groups via `groupId` so that two services (in the same microservice) consuming from the same topic don't interfere. I had a `security-service-impl` in mind where there's a `CredentialsServiceImpl` and a `TokenServiceImpl` both consuming from a fraud topic.